src/lib/components/ui/Card/Card.jsx

```diff
@@ -1,87 +1,49 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import getRootVariantClassName from '../../../helpers/getRootVariantClassName';
 import { withProviderContext } from '../../../provider';
 import styles from './Card.scss';
 
-export const Card = (props) => {
-  const {
-    children,
-    dense,
-    disabled,
-    id,
-    raised,
-    type,
-    variant,
-  } = props;
-
-  const overriddenVariant = type === null ? variant : 'bordered';
-
-  let typeClass = '';
-  if (type === 'success') {
-    typeClass = styles.isTypeSuccess;
-  } else if (type === 'warning') {
-    typeClass = styles.isTypeWarning;
-  } else if (type === 'error') {
-    typeClass = styles.isTypeError;
-  } else if (type === 'help') {
-    typeClass = styles.isTypeHelp;
-  } else if (type === 'info') {
-    typeClass = styles.isTypeInfo;
-  } else if (type === 'note') {
-    typeClass = styles.isTypeNote;
-  }
-
-  let variantClass = '';
-  if (overriddenVariant === 'flat') {
-    variantClass = styles.isVariantFlat;
-  } else if (overriddenVariant === 'bordered') {
-    variantClass = styles.isVariantBordered;
-  }
-
-  let denseClass = '';
-  if (dense) {
-    denseClass = styles.isDense;
-  }
-
-  let disabledClass = '';
-  if (disabled) {
-    disabledClass = styles.isDisabled;
-  }
-
-  let raisedClass = '';
-  if (raised) {
-    raisedClass = styles.isRaised;
-  }
-
-  return (
-    <div
-      className={(`
-        ${styles.root}
-        ${typeClass}
-        ${variantClass}
-        ${denseClass}
-        ${disabledClass}
-        ${raisedClass}
-      `).trim()}
-      id={id}
-    >
-      {children}
-    </div>
-  );
-};
+export const Card = ({
+  bordered,
+  children,
+  dense,
+  disabled,
+  id,
+  raised,
+  variant,
+}) => (
+  <div
+    className={[
+      styles.root,
+      getRootVariantClassName(variant, styles),
+      bordered ? styles.rootBordered : '',
+      dense ? styles.rootDense : '',
+      raised ? styles.rootRaised : '',
+      disabled ? styles.isDisabled : '',
+    ].join(' ')}
+    id={id}
+  >
+    {children}
+  </div>
+);
 
 Card.defaultProps = {
+  bordered: false,
   dense: false,
   disabled: false,
   id: undefined,
   raised: false,
-  type: null,
-  variant: 'flat',
+  variant: null,
 };
 
 Card.propTypes = {
   /**
-   * Slot for individual Card elements that build up the inner layout: CardBody, CardFooter, or
+   * Paint border around the card.
+   */
+  bordered: PropTypes.bool,
```

Comment:
Since `variant` only has visible effect when `bordered=true` I think we can omit the `bordered` prop as it can lead to unexpected result when <Card bordered={false} variant="success" /> will not show any green anywhere.

Having `variant` set to non `null` value would draw the border. Setting `variant` to `null` would draw no border.

Of course this can only work if `variant` is only used to set border color. But since `Card` is such a simple component I think we can make this assumption.

Reply:
I am not sure if there is not something deeper in using those props separately. @adamkudrna?

Reply:
> Since variant only has visible effect when bordered=true

It's no longer the case. I decided to draw border of a `variant` even when `bordered` is not set (and setting it has no effect in such case, the border is already there).

Also, omitting the `bordered` prop is not possible—unless we change the design of _our project_ at the same time.

Current situation in the _project_ is:

- We need flat, border-less cards for large surfaces like the main content area.
- We use bordered `Card` for card listings. The color of the border is neither `light` nor `dark` (see variants), it's just a shade of gray from our gray palette.

We could only get rid of the `bordered` prop if we changed the design to:

1. only use shadow (using the `raised` prop) for card listings, or
2. use `dark` border variant when a border is needed.

From the design point of view, 1) requires heavier shadows to separate white card from white background and 2) would make the UI too cluttered with borders.

So I think we need to keep the `bordered` option as is.

Reply:
We decided to get rid of the `bordered` prop and make the border mandatory, yet customizable through theming options.